src/connect/context/ItemsContext.tsx

```diff
@@ -0,0 +1,72 @@
+import { TreeItem, BaseTreeItem, UITreeItem } from '@/connect/components/tree-view-item';
+import React, { useContext, useState } from 'react';
+
+export interface TreeItemContext {
+    items: TreeItem[];
+    uiState: UITreeItemState;
+    baseItems: BaseTreeItem[];
+    virtualItems: BaseTreeItem[];
+    setUIState: React.Dispatch<React.SetStateAction<UITreeItemState>>;
+    setBaseItems: React.Dispatch<React.SetStateAction<BaseTreeItem[]>>;
+    setVirtualItems: React.Dispatch<React.SetStateAction<BaseTreeItem[]>>;
+    setItems: React.Dispatch<React.SetStateAction<BaseTreeItem[]>>;
+}
+
+const defaultTreeItemContextValue: TreeItemContext = {
+    items: [],
+    uiState: {},
+    baseItems: [],
+    virtualItems: [],
+    setItems: () => {},
+    setUIState: () => {},
+    setBaseItems: () => {},
+    setVirtualItems: () => {},
+};
+
+export const ItemsContext = React.createContext<TreeItemContext>(
+    defaultTreeItemContextValue,
+);
+
+export interface ItemsContextProviderProps {
+    children?: React.ReactNode;
+    items?: BaseTreeItem[];
+}
+
```

Comment:
why not do this instead of disabling the rule

```
export type UITreeItemState = Record<string, UITreeItem>;
```

Reply:
[4b1a171](https://github.com/powerhouse-inc/design-system/pull/94/commits/4b1a171e9b9f348177a8710eb0113bfc32a737ad)